azurerm/internal/services/cosmos/resource_arm_cosmosdb_gremlin_graph.go

```diff
@@ -192,12 +193,11 @@ func resourceArmCosmosDbGremlinGraphCreate(d *schema.ResourceData, meta interfac
 				return fmt.Errorf("Error checking for presence of creating Cosmos Gremlin Graph %s (Account: %s, Database: %s): %+v", name, account, database, err)
 			}
 		} else {
-			id, err := azure.CosmosGetIDFromResponse(existing.Response)
-			if err != nil {
-				return fmt.Errorf("Error getting import ID for Cosmos Gremlin Graph '%s' (Account: %s, Database: %s)", name, account, database)
+			if existing.ID != nil && *existing.ID != "" {
```

Comment:
we want to raise an error if there's already an existing resource (e.g. an ID is returned) so this wants to become:

```suggestion
			if existing.ID == nil || *existing.ID == "" {
```